extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/health/DataSourceHealthCheck.java

```diff
@@ -0,0 +1,62 @@
+package io.quarkus.agroal.runtime.health;
+
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.spi.Bean;
+import javax.sql.DataSource;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Readiness;
+
+import io.quarkus.arc.Arc;
+
+@Readiness
+@ApplicationScoped
+public class DataSourceHealthCheck implements HealthCheck {
+    private static final String DEFAULT_DS = "__default__";
+    private Map<String, DataSource> dataSources = new HashMap<>();
+
+    @PostConstruct
+    protected void init() {
+        Set<Bean<?>> beans = Arc.container().beanManager().getBeans(DataSource.class);
+        for (Bean<?> bean : beans) {
+            if (bean.getName() == null) {
+                // this is the default DataSource: retrieve it by type
+                DataSource defaultDs = Arc.container().instance(DataSource.class).get();
+                dataSources.put(DEFAULT_DS, defaultDs);
+            } else {
+                DataSource ds = (DataSource) Arc.container().instance(bean.getName()).get();
+                dataSources.put(bean.getName(), ds);
+            }
+        }
+    }
+
+    @Override
+    public HealthCheckResponse call() {
+        HealthCheckResponseBuilder builder = HealthCheckResponse.named("Database connection(s) health check").up();
+        int cpt = 1;
+        for (Map.Entry<String, DataSource> dataSource : dataSources.entrySet()) {
+            try {
+                boolean valid = dataSource.getValue().getConnection().isValid(0);
+                if (!valid) {
+                    String data = dataSource.getKey() == DEFAULT_DS ? "validation check failed for the default DataSource"
```

Comment:
String compare should always use `equals`.

Reply:
Arrrhgggg such a newbie error! I'm really ashame as I think I didn't do this error since at least 10 yers !

Reply:
fixed

Reply:
For the record, it worked because it's the same string constant that is defined on the class and put as hashmap key so it's valid to use `==` but I'm agree that this should not be used!